kpops/components/base_components/helm_app.py

```diff
@@ -0,0 +1,166 @@
+from kpops.utils.colorify import magentaify
+from kpops.utils.docstring import describe_attr
+
+log = logging.getLogger("HelmApp")
+
+
+class HelmApp(KubernetesApp):
+    """Kubernetes app managed through Helm with an associated Helm chart.
+
+    :param repo_config: Configuration of the Helm chart repo to be used for
+        deploying the component, defaults to None this means that the command "helm repo add" is not called and Helm
+        expects a path to local Helm chart.
+    :param version: Helm chart version, defaults to None
+    """
+
+    repo_config: HelmRepoConfig | None = Field(
+        default=None,
+        description=describe_attr("repo_config", __doc__),
+    )
+    version: str | None = Field(
+        default=None,
+        description=describe_attr("version", __doc__),
+    )
+
+    @cached_property
+    def helm(self) -> Helm:
+        """Helm object that contains component-specific config such as repo."""
+        helm = Helm(self.config.helm_config)
+        if self.repo_config is not None:
+            helm.add_repo(
+                self.repo_config.repository_name,
+                self.repo_config.url,
+                self.repo_config.repo_auth_flags,
+            )
+        return helm
+
+    @cached_property
+    def helm_diff(self) -> HelmDiff:
+        """Helm diff object of last and current release of this component."""
+        return HelmDiff(self.config.helm_diff_config)
+
+    @cached_property
+    def dry_run_handler(self) -> DryRunHandler:
+        helm_diff = HelmDiff(self.config.helm_diff_config)
+        return DryRunHandler(self.helm, helm_diff, self.namespace)
+
+    @property
+    def helm_release_name(self) -> str:
+        """The name for the Helm release. Can be overridden."""
+        return self.full_name
+
+    @property
+    def helm_chart(self) -> str:
+        """Return component's Helm chart."""
+        msg = (
+            f"Please implement the helm_chart property of the {self.__module__} module."
+        )
+        raise NotImplementedError(msg)
+
+    @property
+    def helm_flags(self) -> HelmFlags:
+        """Return shared flags for Helm commands."""
+        auth_flags = self.repo_config.repo_auth_flags.dict() if self.repo_config else {}
+        return HelmFlags(
+            **auth_flags,
+            version=self.version,
+            create_namespace=self.config.create_namespace,
+        )
+
+    @property
+    def template_flags(self) -> HelmTemplateFlags:
+        """Return flags for Helm template command."""
+        return HelmTemplateFlags(
+            **self.helm_flags.dict(),
+            api_version=self.config.helm_config.api_version,
+        )
+
+    @override
+    def template(self) -> None:
+        stdout = self.helm.template(
+            self.helm_release_name,
+            self.helm_chart,
+            self.namespace,
+            self.to_helm_values(),
+            self.template_flags,
+        )
+        print(stdout)
```

Comment:
Didn't we want to change the print statement here with a better solution? 🤔 
Something like: https://github.com/bakdata/kpops/blob/main/kpops/pipeline_generator/pipeline.py#L249